__tests__/commands/_helpers.js

```diff
@@ -79,13 +79,24 @@ export async function run<T, R>(
 
   const reporter = new Reporter({stdout, stderr: stdout});
 
-  const dir = path.join(fixturesLoc, name);
-  const cwd = path.join(
-    os.tmpdir(),
-    `yarn-${path.basename(dir)}-${Math.random()}`,
-  );
-  await fs.unlink(cwd);
-  await fs.copy(dir, cwd, reporter);
+  let cwd;
+  if (fixturesLoc) {
+    // if fixture loc is not set create a tmp dir so that we don't copy CWD during test run
+    const dir = path.join(fixturesLoc, name);
+    cwd = path.join(
+      os.tmpdir(),
+      `yarn-${path.basename(dir)}-${Math.random()}`,
+    );
+    await fs.unlink(cwd);
+    await fs.copy(dir, cwd, reporter);
+  } else {
+    cwd = path.join(
+      os.tmpdir(),
+      `yarn-${Math.random()}`,
```

Comment:
Does Node.js provide a way to call the system's native temporary directory creation function (such as [mkdtemp](http://man7.org/linux/man-pages/man3/mkdtemp.3.html) on GNU/Linux or [GetTempFileName](https://msdn.microsoft.com/en-us/library/windows/desktop/aa364991(v=vs.85).aspx) on Windows)? The issue with creating temporary directories like this is that two different processes (for example, running different tests in parallel) could generate the same random number.

Failing that, a GUID would be better too.